### PR TITLE
Implement basic memory loads and stores

### PR DESCRIPTION
**Pull request by aviansie-ben**
This changeset implements loads and stores for i32, i64, f32, and f64 values and adds tests for these opcodes.

### CODE REVIEW

**Review by Leonardo2718 — CHANGES_REQUESTED** [review]
LGTM! I will merge this after the merge conflicts are resolved.

**Comment by aviansie-ben** [comment]
Merge conflicts have been resolved.

**Review by Leonardo2718 — APPROVED** [review]

**Pull request merged**
